realm/src/main/java/io/realm/RealmBaseAdapter.java

```diff
@@ -89,7 +94,20 @@ public long getItemId(int i) {
      * @param realmResults the new RealmResults coming from the new query.
      */
     public void updateRealmResults(RealmResults<T> realmResults) {
+        if (listener != null) {
+            if (this.realmResults != null) {
+                this.realmResults.getRealm().removeChangeListener(listener);
+            }
+            if (realmResults != null) {
```

Comment:
Using both ```this.realmResults``` and ```realmResults``` is a bit confusing to read. A different name for the method variable would increase readability. Perhaps just "results" or "queryResults"?